README.md

```diff
@@ -160,21 +143,104 @@ try {
 }
 ```
 
+Some kinds of dependency fallbacks may be handled by [import maps](https://github.com/wicg/import-maps), but even with the support of import maps, [some scenarios](https://github.com/WICG/import-maps/blob/master/README.md#alternating-logic-based-on-the-presence-of-a-built-in-module) which are outside of the declaratively handled set would still benefit from top-level `await`.
+
+### WebAssembly Modules
+
+WebAssembly Modules are "compiled" and "instantiated" in a logically asynchronous way, based on their imports: Some WebAssembly implementations do nontrivial work at either phase, which is important to be able to shunt off into another thread. To integrate with the JavaScript module system, they will need to do the equivalent of a top-level await. See the [WebAssembly ESM integration proposal](https://github.com/webassembly/esm-integration) for more details.
+
+## Semantics as desugaring
+
+Top-level `await` makes importing a module automatically block on any top-level `await`s. You can think of it like, each module exports a Promise, and after all the `import` statements, but before the rest of the module, the Promises are all `await`ed:
+
+```mjs
+import { a } from './a.mjs'
+import { b } from './b.mjs'
+import { c } from './c.mjs'
+
+console.log(a, b, c);
+```
+
+would be equivalent to
+
+```mjs
+import aPromise, { a } from './a.mjs'
+import bPromise, { b } from './b.mjs'
+import cPromise, { c } from './c.mjs'
+
+Promise.all([aPromise, bPromise, cPromise]).then(() => {
+
+console.log(a, b, c);
+
+});
+```
+
+Modules `a.mjs`, `b.mjs`, and `c.mjs` would all execute in order up until the first await in each of them; we then wait on all of them to resume and finish evaluating before continuing.
+
 ### FAQ
 
 #### Isn't top-level `await` a footgun?
 
-If you have seen [the gist](https://gist.github.com/Rich-Harris/0b6f317657f5167663b493c722647221) you likely have heard this critique before. My hope is that as a committee we can weigh the pros / cons of the various approaches and determine if the benefits of the feature outweigh the risks.
+If you have seen [the gist](https://gist.github.com/Rich-Harris/0b6f317657f5167663b493c722647221) you likely have heard this critique before.
+
+Some responses to some of the top concerns here:
+
+##### Will top-level `await` cause developers to make their code block longer than it should?
+
+It's true that top-level `await` gives developers a new tool to make their code wait. Our hope is that proper developer education can ensure that the semantics of top-level `await` are well-understood, so that people know to use it just when they intend that importers should block on it.
+
+We've seen this work well in the past. For example, it's easy to write code with async/await that serializes two tasks that could be done in parallel, but a deliberate developer education effort has popularized the use of `Promise.all` to avoid this hazard.
```

Comment:
I very much disagree this is popularized; I have to tell people to use `Promise.all` to avoid unnecessarily serializing their code with `await` multiple times a day, every day, since async/await became popular. It's a commonly fired footgun, and education has not yet been sufficient to avoid it.

Reply:
Interesting; maybe file an issue about it? Also seems like an argument about the feature as a whole.